src/main/java/org/wso2/carbon/module/swiftiso20022/MT103ToJsonTransformer.java

```diff
@@ -0,0 +1,149 @@
+
+    private static final Log log = LogFactory.getLog(MT103ToJsonTransformer.class);
+    private static final Gson gson = new Gson();
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        log.debug("Executing MT103ToJsonTransformer to transform MT103 text message to JSON");
+
+        String payload = PayloadHelper.getTextPayload(messageContext);
+
+        if (StringUtils.isBlank(payload)) {
+            log.error("Failed to read the text payload from request.");
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.MISSING_REQUEST_PAYLOAD,
+                    ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            this.handleException(ConnectorConstants.ERROR_MISSING_PAYLOAD, messageContext);
+        }
+
+        try {
+            MT103Message mt103Message = MT103Parser.parse(payload);
+
+            // TODO: validate MT103 message
+            // Current implementation doesn't validate values
+            // and doesn't enforce format on fields with multiple options
+
+            String jsonString = transformJsonObject(gson.toJsonTree(mt103Message).getAsJsonObject());
+
+            ConnectorUtils.appendJsonResponseToMessageContext(messageContext, jsonString);
+
+        } catch (MTMessageParsingException e) {
+            log.error("Failed to parse text payload to MT103 model.", e);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.INVALID_REQUEST_PAYLOAD,
+                    e.getMessage());
+            this.handleException(e.getMessage(), messageContext);
+        } catch (AxisFault e) {
+            log.error(ConnectorConstants.PROCESSING_ERROR, e);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.SERVER_ERROR,
+                    ConnectorConstants.PROCESSING_ERROR);
+            throw new ConnectException(e, ConnectorConstants.PROCESSING_ERROR);
+        }
+    }
+
+    /**
+     * Method to transform JSON object of {@link MT103Message}.
+     * Replace amount strings as number.
+     * Remove no letter option.
+     *
+     * @param mt103Message JSON conversion of {@link MT103Message} using Gson.
+     * @return Transformed JSON object as a String.
+     */
+    private static String transformJsonObject(JsonObject mt103Message) {
+
+        JsonObject textBlock = mt103Message.getAsJsonObject(MT103Constants.TEXT_BLOCK_KEY);
+
+        replaceAmountAsNumber(textBlock);
+
+        JsonTransformationUtils.removeNoLetterOption(textBlock);
+
+        mt103Message.add(MT103Constants.TEXT_BLOCK_KEY, textBlock);
+
+        return mt103Message.toString();
+    }
+
+    /**
+     * Method to replace amount strings as numbers from MT103 model JSON.
+     *
+     * @param textBlock JSON conversion of the model {@link MT103Message} (converted using Gson).
+     */
+    private static void replaceAmountAsNumber(JsonObject textBlock) {
```

Comment:
Shall we move these common methods to common util class as discussed during the code review